### ISSUE TITLE
[FR]: Endless spool

### ISSUE THREAD

**Issue by allenrowand** [3DCoded/3MS#5]
### What feature do you want to see?

Was wondering if there was any plan to add endless spool, so that (for example) if T0 runs out of filament 3MS would pause, unload T0, load T1, and resume. I'm assuming the best way to implement that would be a runout sensor before each 3MS extruder…

Thanks!

**Comment by 3DCoded** [comment]
Experimental implementation is complete, and documentation is [here](https://3dcoded.github.io/3MS/experimental/endless/). Let me know if anything needs to be changed.

**Comment by allenrowand** [comment]
Would like to talk to you about the project, can you email me at allen@west3d.com?

**Comment by 3DCoded** [comment]
I've implemented support for Endless Spool for another type of filament sensor setup: a single filament sensor setup, which is what my setup uses. 

After some testing and evaluating the internal logic, a single filament sensor setup should be more reliable than a multi-filament sensor setup (the multi-filament sensor setup requires unloading the empty spool, which could lead to tip shaping issues, while the single filament sensor setup simply rams the new filament into the old without any unload required).

I've updated the documentation with the new recommendations and settings.
